core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -146,4 +151,44 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
         );
       });
   }
+
+  /* Pagination handler */
+  onPageIndexChange(pageIndex: number): void {
+    this.currentPageIndex = pageIndex;
+    this.changePaginatedExecutions();
+  }
+
+  onPageSizeChange(pageSize: number): void {
+    this.pageSize = pageSize;
+    this.changePaginatedExecutions();
+  }
+
+  /**  Render executions when changing page
+   * To-do: create WorkflowExecutionEntry to handle pageIndex, or
+   * add page attributes to workflowExecutionService with function
+   * getPaginatedExecutions()
+   */
+  changePaginatedExecutions() {
+    if (this.workflow.wid === undefined) {
+      return;
+    }
+    this.workflowExecutionsService
+      .retrieveWorkflowExecutions(this.workflow.wid)
+      .pipe(untilDestroyed(this))
+      .subscribe(data => {
+        this.workflowExecutionsList = [];
+
+        let limit = this.currentPageIndex * this.pageSize;
+        let iter = (this.currentPageIndex - 1) * this.pageSize;
+        for (let i = iter; i < limit; i++) {
```

Comment:
this is a very expensive operation. Why can't you just display within the two specified indexes.

Reply:
It actually calls all items between two indexes. It will start from the first index of a page to last index of the same page.